src/sentry/api/endpoints/source_map_debug.py

```diff
@@ -147,4 +154,55 @@ def get(self, request: Request, project: Project, event_id: str) -> Response:
                 }
             )
 
+        release_artifacts = ReleaseFile.objects.filter(release_id=release.id)
+
+        full_matches = [
+            artifact
+            for artifact in release_artifacts
+            if urlparse(artifact.name).path == urlparts.path
+        ]
+        partial_matches = [
+            artifact
+            for artifact in release_artifacts
+            if artifact.name.endswith(urlparts.path.split("/")[-1])
+        ]
+
+        if len(full_matches) == 0:
+            if len(partial_matches) > 0:
+                return Response(
+                    {
+                        "errors": [
+                            SourceMapProcessingIssue(
+                                SourceMapProcessingIssue.PARTIAL_MATCH,
+                                data={"absPath": abs_path, "partialMatch": partial_matches[0].name},
+                            ).get_api_context()
+                        ]
+                    }
+                )
+            return Response(
+                {
+                    "errors": [
+                        SourceMapProcessingIssue(
+                            SourceMapProcessingIssue.NO_URL_MATCH,
+                            data={"absPath": abs_path},
+                        ).get_api_context()
+                    ],
+                }
+            )
+
+        artifact = full_matches[0]
+
+        dist = Distribution.objects.get(release=release, name=event.dist)
```

Comment:
@roggenkemper we should have a try/catch here in case this returns 0 results, right?